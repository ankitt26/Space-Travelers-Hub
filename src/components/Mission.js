/*eslint-disable*/
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../Redux/missions/missionsSlice';

function Mission({
  mission_id, mission_name, description, reserved,
}) {
  const dispatch = useDispatch();

  function handleClick(e, mission_id, reserved) {
    e.preventDefault();
    if(reserved) {
      dispatch(leaveMission({mission_id}));
    } else {
      dispatch(joinMission({mission_id}));
    }
  }
  
  return (
    <div className="mission">
      <div className="mission-name">
        <p>{mission_name}</p>
      </div> 
      <div className="description">
        <p>{description}</p>
      </div>
      <div className="status">
        {reserved === true && <p>Active Member</p>}
        {reserved === false && <p>Not a Member</p>}
      </div>
      <div className="join-leave">
        <button key={mission_id} type="button" className="join-btn" onClick={(e) => handleClick(e, mission_id, reserved)}>{reserved ? 'Leave Mission' : 'Join Mission'}</button>
      </div>
    </div>
  );
}

Mission.propTypes = {
  mission_name: PropTypes.string,
  mission_id: PropTypes.string,
  description: PropTypes.string,
  reserved: PropTypes.bool.isRequired,
};

Mission.defaultProps = {
  mission_name: '',
  mission_id: '',
  description: '',
  reserved: false,
};

export default Mission;
