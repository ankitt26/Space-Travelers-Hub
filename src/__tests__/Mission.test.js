import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import Mission from '../components/Mission'
import Store from "../Redux/store";
import { MemoryRouter } from "react-router";

it("MyProfile component test", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={Store}>
          <Mission />
        </Provider>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

