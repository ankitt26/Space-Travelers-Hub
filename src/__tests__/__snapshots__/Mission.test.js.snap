// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component test 1`] = `
<div
  className="mission"
>
  <div
    className="mission_name"
  >
    <p />
  </div>
  <div
    className="description"
  >
    <p />
  </div>
  <div
    className="status"
  >
    <p
      className="not_member"
    >
      NOT A MEMBER
    </p>
  </div>
  <div
    className="join_leave"
  >
    <button
      className="join_btn"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </div>
</div>
`;
